### PR TITLE
Configure included editor components per field, add optional minimal height

### LINKED ISSUE #1788: Limit the choice of block level component in the Markdown editor
Is it possible to customize the editor ui bar of the markdown widget to limit the choice of block level component in the Markdown editor?

I know you can do:
` -{label: "Markdown", name: "markdown", widget: "markdown", buttons: ["bold", "italic"]`

But, it will be perfect to have the possibility to add:
`
-{label: "Markdown", name: "markdown", widget: "markdown", components: ["youtube", "image", "vimeo"]}
`

### PR DESCRIPTION
**Pull request by erquhart**
**Summary**

 
Resolves #1788.
- Allows specific editor components to be enabled per markdown widget via `editorComponents` field property, with an empty list removing the dropdown and button completely. Important when you want markdown content to have no block level entities.
- Allows minimal height markdown widget via `minimal` field property. When authoring a small amount of markdown, often with inline content only, the 300px minimal height is overkill. This becames a bigger problem with a list of markdown widgets.

### CODE REVIEW

File: packages/netlify-cms-widget-markdown/src/MarkdownControl/RawEditor.js [hunk 1]
```diff
@@ -13,19 +13,17 @@ import { markdownToHtml } from '../serializers';
...
-    border-top-right-radius: 0;
-    border-top: 0;
-    margin-top: -${editorStyleVars.stickyDistanceBottom};
-  `,
-};
+const rawEditorStyles = ({ minimal }) => `
+  position: relative;
+  overflow: hidden;
+  overflow-x: auto;
+  min-height: ${minimal ? 'auto' : lengths.richTextEditorMinHeight};
```
Comment (erquhart) [inline packages/netlify-cms-widget-markdown/src/MarkdownControl/RawEditor.js]: Besides wrapping these styles in a function, this is the only changed line here.

File: packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js [hunk 2]
```diff
@@ -86,9 +87,9 @@ export default class Toolbar extends React.Component {
     t: PropTypes.func.isRequired,
   };
 
-  isHidden = button => {
+  isVisible = button => {
     const { buttons } = this.props;
-    return List.isList(buttons) ? !buttons.includes(button) : false;
+    return !List.isList(buttons) || buttons.includes(button);
```
Comment (erquhart) [inline packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js]: This switch from `isHidden` to `isVisible` is unrelated cleanup, toolbar buttons were handling their own visibility.

File: website/content/docs/widgets/markdown.md [hunk 3]
```diff
@@ -12,7 +12,9 @@ _Please note:_ If you want to use your markdown editor to fill a markdown file c
 - **Data type:** markdown
 - **Options:**
   - `default`: accepts markdown content
-  - `buttons`: an array of strings representing the formatting buttons to display, all buttons shown by default. Buttons include: `bold`, `italic`, `code`, `link`, `heading-one`, `heading-two`, `heading-three`, `heading-four`, `heading-five`, `heading-six`, `quote`, `code-block`, `bulleted-list`, and `numbered-list`.
+  - `minimal`: accepts a boolean value, `false` by default. Sets the widget height to minimum possible.
+  - `buttons`: an array of strings representing the formatting buttons to display (all shown by default). Buttons include: `bold`, `italic`, `code`, `link`, `heading-one`, `heading-two`, `heading-three`, `heading-four`, `heading-five`, `heading-six`, `quote`, `code-block`, `bulleted-list`, and `numbered-list`.
+  - `editor-components`: an array of strings representing the names of editor components to display (all shown by default). The `image` editor component is included with Netlify CMS by default, but others may be [created and registered](/docs/custom-widgets/#registereditorcomponent).
```
Comment (erezrokah) [inline website/content/docs/widgets/markdown.md]: This should be `field.get('editorComponents')` per the code right?

File: packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js [hunk 4]
```diff
@@ -114,52 +115,59 @@ export default class Toolbar extends React.Component {
       hasMark = () => {},
       hasInline = () => {},
       hasBlock = () => {},
+      editorComponents,
       t,
     } = this.props;
+    const isVisible = this.isVisible;
+    const showEditorComponents = !editorComponents || editorComponents.size > 1;
```
Comment (erezrokah) [inline packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js]: Should be `editorComponents.size >= 1`

File: packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js [hunk 5]
```diff
@@ -114,52 +115,59 @@ export default class Toolbar extends React.Component {
...
+    const showPlugin = ({ name }) => (editorComponents ? editorComponents.includes(name) : true);
```
Comment (erezrokah) [inline packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js]: I had to change this line to `const showPlugin = ({ id }) => (editorComponents ? editorComponents.includes(id) : true);` to get it to work on my dev env.

Comment (erquhart) [reply packages/netlify-cms-widget-markdown/src/MarkdownControl/Toolbar.js]: Good catch, not sure how this was working for me.
